src/edu/harvard/lib/librarycloud/items/ItemResource.java

```diff
@@ -0,0 +1,148 @@
+* It uses the jersey implementation (see web.xml) of jax-rs api for RESTful web services.
+* Requests of type items/{id} are for an individual item;
+* Requests of type items?<search parameters> are for searching the api;
+* 
+* @author Michael Vandermillen
+*
+*/
+
+@Path ("/v2")
+public class ItemResource {
+	Logger log = Logger.getLogger(ItemResource.class); 
+	ItemDAO itemdao = new ItemDAO();
+	
+	@GET @Path("items/{id}")
+    @JSONP(queryParam = "callback")
+	@Produces ({"application/javascript", MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML + ";qs=0.9"})
+	public ModsType getItem(@PathParam("id") String id) {
+		log.info("getItem called for id: " + id);
+		ModsType modsType = null; 
+		try {
+			modsType = itemdao.getMods(id);
+		} catch (JAXBException je) {
+			System.out.println(je);
+			log.error(je.getMessage());
+		}
+		return modsType;
+	}
+	
+	
+	@GET @Path("items/{id}.dc")
+    @JSONP(queryParam = "callback")
+	@Produces ({"application/javascript", MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML + ";qs=0.9"})
+	public Metadata getDublinCoreItem(@PathParam("id") String id) {
+		log.info("getDublinCoreItem called for id: " + id);
+		ModsType modsType = null;
+		String dcXml = null;
+		Metadata metadata = null;
+		try {
+			modsType = itemdao.getMods(id);
+			String modsXml = itemdao.marshallObject(modsType);
+			dcXml = itemdao.transform(modsXml, Config.getInstance().DC_XSLT); 
+			metadata = itemdao.getDublinCore(dcXml);
+
+		} catch (JAXBException je) {
+			je.printStackTrace();
+			log.error(je);
+		}
+
+		return metadata;
+	}
+
+
+	@GET @Path("items")
+    @JSONP(queryParam = "callback")
+	@Produces ({"application/javascript", MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML + ";qs=0.9"})
+	public SearchResultsMods getSearchResults(@Context UriInfo ui) {
+		log.info("getSearchResults made query: " + "TO DO");
+	    MultivaluedMap<String, String> queryParams = ui.getQueryParameters();
+	    //we don't currently need to use the pathParam
+	    //MultivaluedMap<String, String> pathParams = ui.getPathParameters();
+
+		SearchResultsMods results = null;
+		
+		try {
+			results = itemdao.getModsResults(queryParams);	
+		} catch (JAXBException je) {
+			je.printStackTrace();
+			log.error(je.getMessage());
+		}
+		return results;
+	}
+	
+	
+	@GET @Path("items.dc")
+    @JSONP(queryParam = "callback")
+	@Produces ({"application/javascript", MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML + ";qs=0.9"})
+	public SearchResultsDC getDublinCoreSearchResults(@Context UriInfo ui) {
+		log.info("getDublinCoreSearchResults made query: " + "TO DO");
+		MultivaluedMap<String, String> queryParams = ui.getQueryParameters();
+		SearchResultsDC resultsDC = null;
+		try {
+			resultsDC = itemdao.getDCResults(queryParams);
+		} catch (JAXBException je) {
+			je.printStackTrace();
+			log.error(je.getMessage());
+		} 
+		catch(Exception e) {
+			e.printStackTrace();

```

Comment:
Let's make this ```catch``` block consistent with the others in terms of how it handles errors.